framework/src/wifi_manager/wifi_manager.c

```diff
@@ -145,21 +145,36 @@ static void wifi_status_set(connect_status_e status)
 
 static void wifi_linkup_event_func(void)
 {
-	wifi_manager_cb_s *wifi_cb = g_manager_info.wmcb;
+	wifi_manager_cb_s *wifi_cb = g_manager_callback;
+	wifi_utils_info info;
+
+	wifi_mutex_acquire(w_mutex, WIFI_UTILS_FOREVER);
+
 	if (g_manager_info.mode == STA_MODE) {
 		nvdbg("WIFI CONNECTED AP - STA MODE");
 		wifi_status_set(AP_CONNECTED);
 
 		/* Perform DHCP client */
 		if (start_dhcp_client() != WIFI_UTILS_SUCCESS) {
 			ndbg("DHCP client start failed\n");
+			wifi_mutex_release(w_mutex);
 			return;
 		}
 		if (stop_dhcp_client() != WIFI_UTILS_SUCCESS) {
 			ndbg("DHCP client stop failed\n");
 		}
 
-		if (wifi_cb != NULL && wifi_cb->sta_connected) {
+		wifi_utils_get_info(&info);
+
+		char ip4_add_str[18] = { 0, };
+		wifi_net_ip4_addr_to_ip4_str(info.ip4_address, ip4_add_str);
```

Comment:
I think return value of wifi_utils_get_info & wifi_netip4_addr_to_ip4_str should be checked to prevent null copying

Reply:
I add the logic for checking the result of wifi_utils_get_info.
but not needed for wifi_net_ip4_addr_to_ip4_str().